server/base/src/main/java/org/apache/accumulo/server/tables/TableManager.java

```diff
@@ -250,6 +250,10 @@ public void removeTable(TableId tableId) throws KeeperException, InterruptedExce
       tableStateCache.remove(tableId);
       zoo.recursiveDelete(zkRoot + Constants.ZTABLES + "/" + tableId + Constants.ZTABLE_STATE,
           NodeMissingPolicy.SKIP);
+      String tableConfigPath = zkRoot + Constants.ZTABLES + "/" + tableId + Constants.ZTABLE_CONF;
+      // The line below removes all the watches on the table configuration items from the Zookeeper
+      // server
+      zoo.getZooKeeper().getChildren(tableConfigPath, false);
```

Comment:
When I started working this issue a few weeks back this code had an effect of removing watchers but the subsequent changes in the ZooCache seems to have made it unneeded.  Thanks for motivating me to retry the solution without this modification.

Reply:
I think that effect would have been seen because the watch had been triggered, and this code requests the children without re-setting that watch. However, if you don't care about the return value, simply not retrieving the children at all would have the same effect as calling `getchildren` with `false` for the watch and throwing away the return value. But, omitting the line does less work with the same effect.

The important thing to note is that watches will be removed over time if they are triggered and not re-set, because watches trigger at most once, and then get removed. If we don't re-set them, it has the same effect as it being deleted.

Reply:
Yeah.  But if they never get triggered they will never get deleted.  When a watch is created on a znode that is never created the watch stays around forever on ZooKeeper.  Also, Zookeeper keeps a session string for each service that has an association to a watch.  So, if you have a thousand node cluster, memory is taken to keep track of each services association to a ZooNode that doesn't exist and won't exist again.

Reply:
Yes, I understand, but the context here is which kind of watcher do we make sure is put on the `conf/` node, not the child nodes.

Reply:
We set a data watch on the 'conf' node.  From the ZooKeeper webpage linked above:
>The data for which the watch was set This refers to the different ways a node can change. It helps to think of ZooKeeper as maintaining two lists of watches: data watches and child watches. getData() and exists() set data watches. getChildren() sets child watches. Alternatively, it may help to think of watches being set according to the kind of data returned. getData() and exists() return information about the data of the node, whereas getChildren() returns a list of children. Thus, setData() will trigger data watches for the znode being set (assuming the set is successful). A successful create() will trigger a data watch for the znode being created and a child watch for the parent znode. A successful delete() will trigger both a data watch and a child watch (since there can be no more children) for a znode being deleted as well as a child watch for the parent znode.

I took this line of code out anyways and pushed the change.  This solution does not require that line of code.